File: backend/model/reminder.go
package model

import "time"

type Reminder struct {
	ID              uint       `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID          uint       `gorm:"not null;index" json:"userId"`
	Title           string     `gorm:"size:255;not null" json:"title"`
	Content         string     `gorm:"type:text" json:"content"`
	RemindAt        time.Time  `gorm:"not null;index" json:"remindAt"`
	Notified        bool       `gorm:"default:false;index" json:"notified"`
	RepeatType      string     `gorm:"size:20;default:'none'" json:"repeatType"`
	RepeatInterval  int        `gorm:"default:1" json:"repeatInterval"`
	RepeatEndAt     *time.Time `json:"repeatEndAt"`
	CreatedAt       time.Time  `json:"createdAt"`
	UpdatedAt       time.Time  `json:"updatedAt"`
}

func (Reminder) TableName() string {
	return "reminders"
}

type CreateReminderRequest struct {
	Title          string     `json:"title" binding:"required"`
	Content        string     `json:"content"`
	RemindAt       time.Time  `json:"remindAt" binding:"required"`
	RepeatType     string     `json:"repeatType"`
	RepeatInterval int        `json:"repeatInterval"`
	RepeatEndAt    *time.Time `json:"repeatEndAt"`
}

type UpdateReminderRequest struct {
	Title          string     `json:"title"`
	Content        string     `json:"content"`
	RemindAt       time.Time  `json:"remindAt"`
	RepeatType     string     `json:"repeatType"`
	RepeatInterval int        `json:"repeatInterval"`
	RepeatEndAt    *time.Time `json:"repeatEndAt"`
}

File: frontend/src/service/api/reminder.ts
import { request } from '../request';

export interface ReminderParams {
  title: string;
  content: string;
}

export interface Reminder {
  id: number;
  jobId: number;
  userId: number;
  scheduledAt: string;
  status: string;
  params: ReminderParams;
  repeatType: 'none' | 'daily' | 'weekly' | 'monthly' | 'yearly';
  repeatInterval: number;
  repeatEndAt: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateReminderParams {
  title: string;
  content?: string;
  remindAt: string;
  repeatType?: string;
  repeatInterval?: number;
  repeatEndAt?: string | null;
}

export interface UpdateReminderParams {
  title?: string;
  content?: string;
  remindAt?: string;
  repeatType?: string;
  repeatInterval?: number;
  repeatEndAt?: string | null;
}

export function fetchGetReminders(params: { year: number; month: number }) {
  return request<Reminder[]>({ url: '/api/reminders', method: 'get', params });
}

export function fetchCreateReminder(data: CreateReminderParams) {
  return request<Reminder>({ url: '/api/reminders', method: 'post', data });
}

export function fetchUpdateReminder(id: number, data: UpdateReminderParams) {
  return request<Reminder>({ url: `/api/reminders/${id}`, method: 'put', data });
}

export function fetchDeleteReminder(id: number) {
  return request({ url: `/api/reminders/${id}`, method: 'delete' });
}